PKHeX.Core/Legality/Verifiers/MarkVerifier.cs

```diff
@@ -69,9 +70,64 @@ public static bool IsMarkValid(RibbonIndex mark, PKM pk, IEncounterTemplate enc)
             RibbonIndex.MarkCurry when !IsMarkAllowedCurry(pk, x) => false,
             RibbonIndex.MarkFishing when !IsMarkAllowedFishing(x) => false,
             RibbonIndex.MarkDestiny => false,
+            >= RibbonIndex.MarkCloudy and <= RibbonIndex.MarkMisty => IsWeatherPermitted(mark, x),
             _ => true
         };
 
+        private static bool IsWeatherPermitted(RibbonIndex mark, IEncounterTemplate enc)
+        {
+            var permit = mark.GetWeather8();
+
+            // Encounter slots check location weather, while static encounters check weather per encounter.
+            return enc switch
+            {
+                EncounterSlot8 w => IsSlotWeatherPermitted(permit, w),
+                EncounterStatic8 s => s.Weather.HasFlag(permit),
+                _ => false,
+            };
+        }
+
+        private static bool IsSlotWeatherPermitted(AreaWeather8 permit, EncounterSlot8 s)
+        {
+            var location = s.Location;
+            // If it's not in the main table, it can only have Normal weather.
+            if (!EncounterArea8.WeatherbyArea.TryGetValue(location, out var mainweather))
+                mainweather = AreaWeather8.Normal;
+
+            // Check bleed conditions first.
+            var type = s.SlotType;
+            if (type is SymbolMain or SymbolMain2 or SymbolMain3)
+            {
+                if (EncounterArea8.WeatherBleedSymbol.TryGetValue(location, out var weather) && (weather.HasFlag(permit) || mainweather.HasFlag(permit)))
+                    return true;
+            }
+            if (type == Surfing)
+            {
+                if (EncounterArea8.WeatherBleedSymbolSurfing.TryGetValue(location, out var weather) && (weather.HasFlag(permit) || mainweather.HasFlag(permit)))
+                    return true;
+            }
+            if (type == Sharpedo)
+            {
+                if (EncounterArea8.WeatherBleedSymbolSharpedo.TryGetValue(location, out var weather) && (weather.HasFlag(permit) || mainweather.HasFlag(permit)))
+                    return true;
+            }
+            if (type is HiddenMain or HiddenMain2 or HiddenMain3)
+            {
+                if (EncounterArea8.WeatherBleedHiddenGrass.TryGetValue(location, out var weather) && (weather.HasFlag(permit) || mainweather.HasFlag(permit)))
+                    return true;
+
+                // Allow all tree and fishing encounters.
+                weather = s.Weather;
+                if (weather.HasFlag(AreaWeather8.Shaking_Trees) || weather.HasFlag(AreaWeather8.Fishing))
+                    return true;
+            }
+
+            if (mainweather.HasFlag(permit))
```

Comment:
Not sure why this is checked last:
* Above branches check it when returning true
* Nothing returns false, so checking it first will do no harm?
* Should any of the inners return false?

If possible, check mainweather first, then we don't have to care about it after?